packages/core/src/logs/utils/deepClone.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+const isDate = (object: unknown): object is Date => {
+    const type = {}.toString.call(object).slice(8, -1);
+    return type === 'Date';
+};
+
+const isArray = (object: unknown): object is unknown[] => {
+    const type = {}.toString.call(object).slice(8, -1);
+    return type === 'Array';
+};
+
+const isObject = (object: unknown): object is Record<string, unknown> => {
+    const type = {}.toString.call(object).slice(8, -1);
+    return type === 'Object';
+};
+
+const isSet = (object: unknown): object is Set<unknown> => {
+    const type = {}.toString.call(object).slice(8, -1);
+    return type === 'Set';
+};
+
+const isMap = (object: unknown): object is Map<string, unknown> => {
+    const type = {}.toString.call(object).slice(8, -1);
+    return type === 'Map';
+};
+
+/**
+ * Simple deep clone inspired from https://github.com/angus-c/just/blob/master/packages/collection-clone/index.cjs
+ * Functions and RegExp will be returned as they are.
+ * Maps and Sets are cloned even though they won't be accepted by React Native, just in case they are mutated by the user to avoid side effects.
+ *
+ * We added support for circular references by limiting the number of depth levels to 8.
```

Comment:
Is this aligned with the depth level we support on Native (can't remember tbh)

Reply:
Yes, this is the same limit we use for timings and user extra info on Android

Reply:
I will add some documentation on that

Reply:
This will be added in the docs, but let's first reach out to the docs team to see where to put this documentation